### PR TITLE
docs: webassembly code analysis

### PR DESCRIPTION
**Pull request by rayowang**
**What this PR does**:
webassembly code analysis
**Which issue(s) this PR fixes**:
#427

### CODE REVIEW

**Comment by seeflood** [comment]
I wonder which tool you used to draw these diagrams. These diagrams look very professional, much better than mine 😢

**Comment by seeflood** [comment]
ok. I used processon before, but the diagrams I drawed are too ugly...... 😢

File: docs/zh/blog/code/webassembly/index.md [hunk 1]
```diff
@@ -0,0 +1,607 @@
...
+	if res != WasmResultOk {
+		return res.Int32()
+	}
+
+	return copyIntoInstance(instance, ret, valuePtr, valueSize).Int32()
+}
+```
+以上 Layotto rpc 流程简要说是通过两个虚拟连接借助 Dapr API 和 底层 Mosn 实现 [5],具体可参见前序文章[Layotto源码解析——处理RPC请求](https://mosn.io/layotto/#/zh/blog/code/layotto-rpc/index)，从 Redis 中获取数据可直接阅读 Dapr State 相关代码，在此不一一展开了。
+
+### Faas模式
```
Comment (zhenjunMa) [inline docs/zh/blog/code/webassembly/index.md:568]: Faas -> FaaS

File: docs/zh/blog/code/webassembly/index.md [hunk 2]
```diff
@@ -0,0 +1,607 @@
...
+
+回过头来再看 WASM 的特性：字节码有与机器码相匹敌的性能；沙箱中执行保证良好的隔离性和安全性；编译后跨平台、易分发和加载运行；具备轻量且多语言开发的灵活性，似乎天然的就适合做 Faas。
```
Comment (zhenjunMa) [inline docs/zh/blog/code/webassembly/index.md:570]: Faas -> FaaS

Comment (zhenjunMa) [reply docs/zh/blog/code/webassembly/index.md:570]: 下面几个也都改成FaaS

File: docs/zh/blog/code/webassembly/index.md [hunk 3]
```diff
@@ -0,0 +1,607 @@
...
+		return
+	}
+
+	cmd := exec.Command("layotto", "start", "-c", "/home/docker/config.json")
+	cmd.Start()
+}
+```
+
+## 总结
+Layotto Webassemly 虽然涉及较多 WASM 相关的基础知识，但通过示例由浅入深，循序渐进也不难理解。最后整体看一下 WASM 技术，可以看到它已经被应用到Web前端、Serverless、游戏场景、边缘计算、服务网格等很多领域，甚至就连 Docker 之父 Solomon Hykes 在前不久都表示: "如果 WASM 这个技术在2008年就有的话，我就不搞Docker了"（后来又补充道：Docker 不会被替换，会与 WASM 并肩而行），不管怎么说，WASM 似乎在继 VM 和 Container 之后，正在成为更轻量及性能更好的云原生技术而被应用到更多的领域，与此同时，相信在 Mosn 社区的推动以及 Layotto 的继续探索中 WASM 也会有更多使用场景和用户，至此 Layotto Webassemly 相关源码分析就完了，鉴于时间和篇幅，没有进行一些更全面和深入的剖析，如有纰漏之处，欢迎指正，联系方式：rayo.wangzl@gmail.com。
```
Comment (zhenjunMa) [inline docs/zh/blog/code/webassembly/index.md:599]: Webassemly -> WebAssembly

**Review by zhenjunMa — CHANGES_REQUESTED** [review]

**Review by zhenjunMa — DISMISSED** [review]
Thanks!

**Review by seeflood — COMMENTED** [review]
@rayowang Hi , I uploaded the images to CDN and submitted a PR to your branch
Please review https://github.com/rayowang/layotto/pull/1

**Comment by seeflood** [comment]
It's weird that this PR didn't update automatically after merging https://github.com/rayowang/layotto/pull/1 😢 
@rayowang Could u please update this PR

**Comment by rayowang** [comment]
> It's weird that this PR didn't update automatically after merging [rayowang#1](https://github.com/rayowang/layotto/pull/1) 😢 @rayowang Could u please update this PR

I was wondering about this problem just now, and later found out that I have two similar branch names, so the branch of this pr is wasm_code_analysis, not webassembly_code_analysis.  So now, do I cherry-pick the changes or do you pr to the wasm_code_analysis branch?

**Comment by seeflood** [comment]
@rayowang OK you can cherry pick those changes

**Comment by rayowang** [comment]
> @rayowang OK you can cherry pick those changes

ok, done.

**Review by seeflood — APPROVED** [review]
lgtm. Thanks!

**Review by zhenjunMa — APPROVED** [review]

**Pull request merged**